prov/usnic/src/usdf_fabric.c

```diff
@@ -745,6 +745,9 @@ usdf_getinfo(uint32_t version, const char *node, const char *service,
 	src = NULL;
 	dest = NULL;
 
+	if (!FI_VERSION_GE(fi_version(), version))
+		return -FI_ENODATA;
+
```

Comment:
The framework should have enough data to provide this sort of check for all providers.  Also, I have a patch against all the providers to update how they set their version.